ambry-frontend/src/main/java/com/github/ambry/frontend/AmbryUrlSigningService.java

```diff
@@ -139,7 +140,9 @@ public String getSignedUrl(RestRequest restRequest) throws RestServiceException
         maxUploadSize = chunkUploadMaxChunkSize;
         // They also have a non-optional blob TTL to ensure that chunks that were not stitched within a reasonable time
         // span are cleaned up.
-        argsForUrl.put(RestUtils.Headers.TTL, chunkUploadInitialChunkTtlSecs);
+        if (RestUtils.getLongHeader(args, RestUtils.Headers.TTL, false) == null) {
+          argsForUrl.put(RestUtils.Headers.TTL, chunkUploadInitialChunkTtlSecs);
```

Comment:
Minor: Just wondering if we should have an `else` case to set the client provided TTL value in the URL arguments? Also, would it be helpful to check for Max TTL and throw error here itself (i.e. during generation of Signed URL) instead of doing it in `checkUploadRequirements` which seem to hit when client tries to upload the blob using the URL. It might probably let client know that his arguments were invalid quicker?

Just to note, it looks like in LI implementation of URLSigningService, above comments seem to be already handled. So, we might only need it for completeness in OSS code.